### PR TITLE
Fixed broken/deprecated links

### PR DESCRIPTION
**Pull request by PhilipSchmid**
Updated links in documentation, the Helm chart and code comments which returned a 302 or even a 404.

### CODE REVIEW

File: vendor/github.com/cilium/ebpf/map.go [hunk 1]
```diff
@@ -1090,7 +1090,7 @@ func (m *Map) Clone() (*Map, error) {
 // You can Clone a map to pin it to a different path.
 //
 // This requires bpffs to be mounted above fileName.
-// See https://docs.cilium.io/en/stable/concepts/kubernetes/configuration/#mounting-bpffs-with-systemd
+// See https://docs.cilium.io/en/stable/network/kubernetes/configuration/#mounting-bpffs-with-systemd
```
Comment (tklauser) [inline vendor/github.com/cilium/ebpf/map.go]: This file (and all others in `vendor/`) are vendored from other packages. These files need to be changed in the respective upstream repository (i.e. `github.com/cilium/ebpf` and `k8s.io/api/core/v1`) first and then vendored back into this repo.

**Review by tklauser — CHANGES_REQUESTED** [review]

**Review by qmonnet — APPROVED** [review]
Thanks! Aside from the changes in `vendor/`, as reported by Tobias, this looks good to me.

Out of curiosity, how did you find these broken links? Did you run some link checker on the code base?

We have one in the docs build system but it's broken and disabled, we should look into it one day when we have cycles (https://github.com/cilium/cilium/issues/10601, [also this link](https://github.com/cilium/cilium/blob/master/Documentation/check-build.sh#L151)). Cc @zacharysarah (just FYI).

**Comment by qmonnet** [comment]
Can you please rebase your changes to get rid of the merge commit?

File: vendor/github.com/cilium/ebpf/map.go (hunk 1 above)
Comment (PhilipSchmid) [reply vendor/github.com/cilium/ebpf/map.go]: Thanks for the info. https://github.com/cilium/ebpf/pull/942

**Edit:** All `vendor/` changes are now removed.

**Comment by PhilipSchmid** [comment]
> Out of curiosity, how did you find these broken links? Did you run some link checker on the code base?

I ran into this broken link here: https://github.com/cilium/cilium/blob/94755dbe5e4dc3ab74e627b7aa164da1634abd47/install/kubernetes/cilium/values.yaml#L1468
.. And because I already run into other broken links in the past (mainly inside the Helm values file), it bugged me so much that I simply took the time to manually "search & replace" all links. An automated link checker would be awesome. I mean, a returned non-200 return code doesn't need to break the whole build, but at least print a warning IMO.

I now also rebased and squashed it.

**Review by tklauser — APPROVED** [review]

**Comment by qmonnet** [comment]
There is no code change (only docs, comments) so I'm not triggering the full CI run.

I see Philip removed some of the review assignments, presumably by accident (otherwise, please don't without stating the motivation). But I think the clean-up is simple enough that we don't need to take it through additional reviews.

Marking as ready-to-merge.

**Review by sayboras — APPROVED** [review]
:heavy_check_mark:

**Comment by PhilipSchmid** [comment]
@qmonnet Sorry for that, I didn't remove them intentionally. Maybe they got removed somehow when I requested another review from Tobias. I'll keep an eye on that next time.

**Pull request merged**